adapters/http-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -631,28 +627,27 @@ private void doUploadMessage(
                     return getTimeUntilDisconnect(tenantTracker.result(), ttdParam)
                             .onSuccess(effectiveTtd -> Optional.ofNullable(effectiveTtd)
                                     .ifPresent(v -> TracingHelper.TAG_DEVICE_TTD.set(currentSpan, v)));
-                });
-        final Future<CommandConsumer> commandConsumerTracker = ttdTracker
-                .compose(ttd -> createCommandConsumer(
-                        ttd,
-                        tenantTracker.result(),
-                        deviceId,
-                        gatewayId,
-                        ctx.getRoutingContext(),
-                        responseReady,
-                        currentSpan));
-
-        commandConsumerTracker
-        .compose(ok -> {
+                })
+                .map(ttd -> ttd == null || ttd <= 0 ? null : ttd); // non positive ttd are made null - no ttd
```

Comment:
I think handling of the "negative ttd" case in the HTTP adapter and the CoAP adapter could be improved in general. I could for example imagine letting a value of "-1" mean that the maximum allowed ttd value is used. Changes in `getTimeUntilDisconnect` should make the above line unnecessary.
But this could be done in a separate PR.

Reply:
should be discussed I suppose. getTimeUntilDisconnect could return Optional<Integer> which if present would always be a vaild timeout.
maybe in different PR